jbt-it/neomdb
# #247 setup neomdb deployment

Added docker-compose files for testing, development and production.
Added nginx config files.
Added /api prefix to all backend routes.
Added documentation about deployment

**Prerequisites:**
- Go to the deployment_files folder in the development folder of the neoMDB in teams and copy the following files/folders into ./deployment
  - env.production (rename it in vsc to **.**env.production)
  - env.development (rename it in vsc to **.**env.development)
  - env.testing (rename it in vsc to **.**env.testing)
  - The secrets folder (./secrets)
- Go to the client_files folder in the development folder of the neoMDB in teams and copy the following files into ./client
  - env.production (rename it in vsc to **.**env.production)
  - env.development (rename it in vsc to **.**env.development)
  - env.testing (rename it in vsc to **.**env.testing)
- Run the following command in ./client: `npm run rerun-prettier` (this should only throw errors regarding json files, which can be ignored) -> This command is necessary, because the linebreaks used in the files are for linux systems and not windows (because the container for deployment use ubuntu linebreaks were changed)

**How to test:**
- First: Check if the local web app functions as expected
- Second: Read DEPLOYMENT.md and check if the explanations are understandable
- Third: Run the testing docker-compose file as specified in DEPLOYMENT.md
- Fourth: Go to dev-neomdb.studentische-beratung.de and test if the web app hosted on the server functions as expected
